src/server/import-controller.js

```diff
@@ -14,18 +15,35 @@ const { NotFound } = httpErrors;
 const jsonParser = express.json();
 
 class ImportController {
-  static get inject() { return [ImportService, ExportApiClient, ServerConfig, PageRenderer]; }
+  static get inject() { return [ServerConfig, PageRenderer, ImportService, ExportApiClient, ClientDataMapper]; }
 
-  constructor(importService, exportApiClient, serverConfig, pageRenderer) {
+  constructor(serverConfig, pageRenderer, importService, exportApiClient, clientDataMapper) {
+    this.serverConfig = serverConfig;
     this.exportApiClient = exportApiClient;
     this.importService = importService;
-    this.serverConfig = serverConfig;
     this.pageRenderer = pageRenderer;
+    this.clientDataMapper = clientDataMapper;
   }
 
   registerPages(app) {
-    app.get('/import', needsPermission(permissions.MANAGE_IMPORT), (req, res) => {
-      return this.pageRenderer.sendPage(req, res, 'edit-bundle', 'import');
+    app.get('/imports', needsPermission(permissions.MANAGE_IMPORT), async (req, res) => {
+      const rawBatches = await this.importService.getImportBatches();
+      const batches = await this.clientDataMapper.mapImportBatches(rawBatches, req.user);
+      const importSources = this.serverConfig.importSources.slice();
+
+      return this.pageRenderer.sendPage(req, res, 'edit-bundle', 'imports', { batches, importSources });
+    });
+
+    app.get('/imports/create', needsPermission(permissions.MANAGE_IMPORT), (req, res) => {
+      return this.pageRenderer.sendPage(req, res, 'edit-bundle', 'create-import');
+    });
+
+    app.get('/imports/:key', needsPermission(permissions.MANAGE_IMPORT), async (req, res) => {
```

Comment:
From a data structure perspective we have `key` for documents only, I'd suggest to call this `:batchId` (same for the corresponding variable). Would make the route clear, plus `getImportBatchDetails` expects something called id.

Reply:
Yep, changed, I was going for consistency with the other endpoints but it really doesn't make sense.